tests/cover/test_numerics.py

```diff
@@ -72,40 +71,37 @@ def test_fuzz_decimals_bounds(data):
         assert val.as_tuple().exponent == -places
 
 
-@fails
-@given(decimals())
-def test_all_decimals_can_be_exact_floats(x):
-    assume(x.is_finite())
-    assert decimal.Decimal(float(x)) == x
+def test_all_decimals_can_be_exact_floats():
+    find_any(
+        decimals(),
+        lambda x: assume(x.is_finite()) and decimal.Decimal(float(x)) == x
+    )
 
 
 @given(fractions(), fractions(), fractions())
 def test_fraction_addition_is_well_behaved(x, y, z):
     assert x + y + z == y + x + z
 
 
-@fails
-@given(decimals())
-def test_decimals_include_nan(x):
-    assert not math.isnan(x)
+def test_decimals_include_nan():
+    find_any(decimals(), lambda x: x.is_nan())
 
 
-@fails
-@given(decimals())
-def test_decimals_include_inf(x):
-    assume(not x.is_snan())
-    assert not math.isinf(x)
+def test_decimals_include_inf():
+    find_any(
+        decimals(),
+        lambda x: assume(not x.is_snan()) and x.is_infinite()
```

Comment:
You probably mean `not x.is_nan()` here - the extra `s` restricts the test to signalling nans, and it could therefore pass without finding an infinity.

Reply:
Actually it looks like I mean neither - decimal NaNs don't return True for is_infinite! Have updated the test.

Reply:
...true, they don't.  Merge when the tests pass! 😄